lib/upipe-srt/upipe_srt_receiver.c

```diff
@@ -0,0 +1,1454 @@
+
+                upipe_srt_receiver->rtt_variance *= 3;
+                upipe_srt_receiver->rtt_variance += var;
+                upipe_srt_receiver->rtt_variance /= 4;
+            }
+            ubase_assert(uref_block_unmap(uref, 0));
+            uref_free(uref);
+        } else {
+
+            if (type == SRT_CONTROL_TYPE_HANDSHAKE && upipe_srt_receiver->establish_time == 0 && now > 0) {
+                uint64_t ts = srt_get_packet_timestamp(buf);
+                upipe_srt_receiver->establish_time = now - ts * UCLOCK_FREQ / 1000000;
+            }
+
+            ubase_assert(uref_block_unmap(uref, 0));
+            if (upipe_srt_receiver->control) {
+                upipe_srt_receiver->last_sent = now;
+                upipe_srt_receiver_output_output(upipe_srt_receiver->control, uref, upump_p);
+            } else
+                uref_free(uref);
+        }
+        return;
+    }
+
+    /* data */
+    if (!upipe_srt_receiver->control) {
+        ubase_assert(uref_block_unmap(uref, 0));
+        uref_free(uref);
+        return;
+    }
+
+    uint32_t seqnum = srt_get_data_packet_seq(buf);
+    uint32_t position = srt_get_data_packet_position(buf);
+    bool order = srt_get_data_packet_order(buf);
+    uint8_t encryption = srt_get_data_packet_encryption(buf);
+    bool retransmit = srt_get_data_packet_retransmit(buf);
+    uint32_t num = srt_get_data_packet_message_number(buf);
+    uint32_t ts = srt_get_packet_timestamp(buf);
+    uint8_t kk = srt_get_data_packet_encryption(buf);
+
+    uint8_t aad[16];
+    bool gcm = encryption && upipe_srt_receiver->salt_len == 12;
+    if (gcm)
+        memcpy(aad, buf, 16);
+
+    ubase_assert(uref_block_unmap(uref, 0));
+    uref_block_resize(uref, SRT_HEADER_SIZE, -1); /* skip SRT header */
+    total_size -= SRT_HEADER_SIZE;
+
+    static const uint64_t wrap = UINT64_C(4294967296);
+    uint64_t delta = (wrap + ts - (upipe_srt_receiver->previous_ts % wrap)) % wrap;
+    int32_t d32 = delta;
+    bool discontinuity = upipe_srt_receiver->previous_ts == 0;
+
+    uint64_t latency_us = (upipe_srt_receiver->latency * 1000000) / UCLOCK_FREQ;
+    /* Note: d32 is converted to unsigned implictly */
+    if (!discontinuity && (d32 <= latency_us || -d32 <= latency_us)) {
+        if (d32 <= latency_us) {
+            upipe_srt_receiver->previous_ts += delta;
+            assert(d32 >= 0);
+        } else {
+             /* out of order but not too old or new */
+        }
+    } else {
+        upipe_warn_va(upipe, "clock ref discontinuity %"PRIu64, delta);
+        upipe_srt_receiver->previous_ts = ts;
+        discontinuity = true;
+    }
+
+    upipe_verbose_va(upipe, "Data seq %u (retx %u)", seqnum, retransmit);
+    if (d32 < 0 && !discontinuity)
+        uref_clock_set_cr_prog(uref, (upipe_srt_receiver->previous_ts + d32) * UCLOCK_FREQ / 1000000);
+    else
+        uref_clock_set_cr_prog(uref, (upipe_srt_receiver->previous_ts) * UCLOCK_FREQ / 1000000);
+
+    (void)order;
+    (void)num;
```

Comment:
Is this just to silence unused variable warnings?  I think you should just comment out the declarations instead.